django/db/backends/base/schema.py

```diff
@@ -998,15 +998,19 @@ def _create_index_name(self, table_name, column_names, suffix=""):
         max_length = self.connection.ops.max_name_length() or 200
         # If everything fits into max_length, use that name.
         index_name = '%s_%s_%s' % (table_name, '_'.join(column_names), hash_suffix_part)
-        if len(index_name) <= max_length:
+        if len(index_name.encode('utf-8')) <= max_length:
             return index_name
         # Shorten a long suffix.
         if len(hash_suffix_part) > max_length / 3:
             hash_suffix_part = hash_suffix_part[:max_length // 3]
         other_length = (max_length - len(hash_suffix_part)) // 2 - 1
+        # Shorten joined column names, accounting for multibyte characters.
+        joined_column_names = '_'.join(column_names)
+        while len(joined_column_names.encode('utf-8')) > other_length:
+            joined_column_names = joined_column_names[:-1]
         index_name = '%s_%s_%s' % (
             table_name[:other_length],
-            '_'.join(column_names)[:other_length],
+            joined_column_names[:other_length],
```

Comment:
Do we have to keep cutting to `other_length`?
```suggestion
            joined_column_names,
```